rllib/core/loss_and_optim/rl_loss_and_optim.py

```diff
@@ -0,0 +1,75 @@
+import abc
+from typing import Mapping, Any, List
+
+from ray.rllib.core.rl_module import RLModule
+from ray.rllib.core.rl_module.marl_module import MultiAgentRLModule
+
+class RLLossAndOptim(abc.ABC):
```

Comment:
Just an idea, how about we call the base class RLOptimizer, which will implicitly mean the loss and optimizer, but also things like update_kl, update_target_q_network() etc. that are not really gradient based updates.

Reply:
or RLUpdator? I personally like RLOptimizer better :)

Reply:
RLUpdator is probably better.

Reply:
or RLModuleUpdater

Reply:
eh optimizer is probably better, it implies a "better" update